### PR TITLE
[ATB-216] Reduce cognitive complexity in checkYourEmailService method

### PR DESCRIPTION
**Pull request by chiarapascucci-gds**
## Proposed changes
Currently the checkYourEmailService Interface describes an async function that takes in 8 parameters. This is flagged as a code smell as the cognitive complexity is too high (max number of parameters is 7). This PR changes the interface definition so that the method signature only requires 2 parameters

### What changed
The CheckYourEmailServiceInterface was updated so to take 2 objects as its parameters. Those objects represent the user's input (email, updated info, and otp code) and session details respectively. The way this information is accessed when forming the post request to the relevant API was also changed in line with this restructuring

### Why did it change
Aim is to reduce cognitive complexity of the method

### Issue tracking

- [ATB-216](https://govukverify.atlassian.net/jira/software/c/projects/ATB/boards/241?modal=detail&selectedIssue=ATB-216)

## Testing
Changes were tested during development

### Local:
Tested the changes locally (in Docker container) through existing unit and integration tests. All passed 
216-unit
216-int

### Deployed:
Deployed a new version of the application to the AWS dev environment successfully and manually tested account creation, changing password, account deletion, log in/out. OTP code received to email address when creating an account
Screenshot 2023-02-07 at 09 19 29

### Environment variables or secrets
- [x] No environment variables or secrets were added or changed

### Other considerations
- [ ] Demo to a BA, TA, and the team.
- [ ] Recorded demo shared on [#govuk-accounts-tech](https://gds.slack.com/archives/C011Y5SAY3U)
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[ATB-216]: https://govukverify.atlassian.net/browse/ATB-216?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/check-your-email/check-your-email-service.ts [hunk 1]
```diff
@@ -2,34 +2,29 @@ import { getRequestConfig, Http, http } from "../../utils/http";
...
       },
       getRequestConfig(
-        accessToken,
+        sessionDetails.accessToken,
         [HTTP_STATUS_CODES.NO_CONTENT, HTTP_STATUS_CODES.BAD_REQUEST],
-        sourceIp,
-        persistentSessionId,
-        sessionId,
-        userLanguage
+          sessionDetails.sourceIp,
```
Comment (TaherKapasi) [inline src/components/check-your-email/check-your-email-service.ts:24]: too much whitepace Line 24 - 27

**Comment by TaherKapasi** [comment]
Hi @chiarapascucci-gds - could apply updates and rebase similar to the comments in [this PR](https://github.com/alphagov/di-authentication-account-management/pull/803) - thanks

**Review by TaherKapasi — CHANGES_REQUESTED** [review]

File: src/components/check-your-email/check-your-email-service.ts [hunk 2]
```diff
@@ -2,34 +2,29 @@ import { getRequestConfig, Http, http } from "../../utils/http";
...
+          sessionDetails.persistentSessionId,
+          sessionDetails.sessionId,
+          sessionDetails.userLanguage
       )
     );
```
Comment (KG700) [inline src/components/check-your-email/check-your-email-service.ts:29]: I think using a `sessionDetails` object is a good improvement - maybe we should extend this approach to all the change details components which are calling `getRequestConfig()` for consistency (probably in a separate ticket). If we do this we could also update the `getRequestConfig()` function to accept the parameters differently, for example, possibly using the spread operator: `getRequestConfig(validationStatues, ...sessionDetails)`

**Review by KG700 — APPROVED** [review]

File: src/components/check-your-email/check-your-email-service.ts (hunk 2 above)
Comment (chiarapascucci-gds) [reply src/components/check-your-email/check-your-email-service.ts:29]: yes :) will look into raising a separate ticket

**Review by chiarapascucci-gds — COMMENTED** [review]
ok

**Review by TaherKapasi — APPROVED** [review]

**Review by KG700 — APPROVED** [review]

**Review by KG700 — APPROVED** [review]

**Pull request merged**